### PR TITLE
refactored injecting repositories

### PR DESCRIPTION
**Pull request by Nazar865**

### CODE REVIEW

File: Corzbank.Services/CardService.cs [hunk 1]
```diff
@@ -66,65 +68,46 @@ public async Task<CardDTO> CreateCard(CardModel cardFromRequest)
...
-                return cardForUpdate;
-            }
-
-            return null;
-        }
-
         public async Task<bool> CloseCard(Guid id)
         {
-            var card = GetCardById(id);
+            var card = await GetById(id);
```
Comment (ghost) [inline Corzbank.Services/CardService.cs]: Use _cardRepo.GetQuearyable()

File: Corzbank.Services/DepositService.cs [hunk 2]
```diff
@@ -67,13 +65,11 @@ public async Task<bool> CloseDeposit(Guid id)
         {
             var deposit = await GetDepositById(id);
```
Comment (ghost) [inline Corzbank.Services/DepositService.cs]: Use _depositRepo.GetQueryable()

File: Corzbank.Web/ClientApp/src/app/data/services/card.service.ts [hunk 3]
```diff
@@ -1,36 +1,39 @@
...
+	constructor(
+		private http: HttpClient,
+		@Inject("BASE_API_URL") private baseUrl: string
+	) {}
 
-  getCard(id: Guid) {
-    return this.http.get(this.url + id);
-  }
+	getCards() {
+		return this.http.get(this.url);
```
Comment (ghost) [inline Corzbank.Web/ClientApp/src/app/data/services/card.service.ts]: return type

Comment (ghost) [reply Corzbank.Web/ClientApp/src/app/data/services/card.service.ts]: add in all places

File: Corzbank.Web/ClientApp/src/app/header/wallet/wallet.component.html [hunk 4]
```diff
@@ -52,18 +52,18 @@ <h2>Deposits</h2>
 		</div>
 		<div class="deposits" *ngIf="depositsIsShown"></div>
 	</div>
-	<div *ngIf="cardsIsShown">
+	<div *ngIf="cardsIsShown && cardDataIsShow">
```
Comment (ghost) [inline Corzbank.Web/ClientApp/src/app/header/wallet/wallet.component.html]: cardDataIsDisplayed looks better

File: Corzbank.Web/ClientApp/src/app/header/wallet/wallet.component.ts [hunk 5]
```diff
@@ -21,23 +24,62 @@ export class WalletComponent implements OnInit {
...
-	selectedCard: CardDTO;
+	selectedCard: Card;
+	cardDataIsShow: boolean;
 	cardNumberIsShown: boolean;
 	cvvIsShown: boolean;
 	cardsIsShown: boolean = true;
 	depositsIsShown: boolean;
-	transfersIsShown: boolean;
+	transactionsIsShown: boolean;
+	transferMenuIsShown: boolean;
```
Comment (ghost) [inline Corzbank.Web/ClientApp/src/app/header/wallet/wallet.component.ts]: isDisplayed

**Review by ghost — CHANGES_REQUESTED** [review]

File: Corzbank.Web/ClientApp/src/app/header/wallet/wallet.component.ts (hunk 5 above)
Comment (Nazar865) [reply Corzbank.Web/ClientApp/src/app/header/wallet/wallet.component.ts]: resolved

File: Corzbank.Web/ClientApp/src/app/header/wallet/wallet.component.html (hunk 4 above)
Comment (Nazar865) [reply Corzbank.Web/ClientApp/src/app/header/wallet/wallet.component.html]: resolved

File: Corzbank.Web/ClientApp/src/app/data/services/card.service.ts (hunk 3 above)
Comment (Nazar865) [reply Corzbank.Web/ClientApp/src/app/data/services/card.service.ts]: resolved

File: Corzbank.Services/DepositService.cs (hunk 2 above)
Comment (Nazar865) [reply Corzbank.Services/DepositService.cs]: resolved

File: Corzbank.Services/CardService.cs (hunk 1 above)
Comment (Nazar865) [reply Corzbank.Services/CardService.cs]: resolved

File: Corzbank.Services/DepositService.cs [hunk 6]
```diff
@@ -63,13 +63,17 @@ public async Task<DepositDTO> OpenDeposit(DepositModel deposit)
 
         public async Task<bool> CloseDeposit(Guid id)
         {
-            var deposit = await GetDepositById(id);
+            var deposit = await _depositRepo
+                .GetQueryable()
+                .Include(c => c.Card)
+                .ThenInclude(u => u.User)
```
Comment (ghost) [inline Corzbank.Services/DepositService.cs:69]: ThenInclude should be on the same level as Include

**Review by ghost — CHANGES_REQUESTED** [review]

**Pull request merged**